### DISCUSSION TITLE
Fetch data in onSubmit method before submit

### DISCUSSION THREAD

**Discussion by carlosriveroib** [TanStack/query#5160] [Q&A] [answered]
I need to fetch data that depends on some fields in the form because I need that to make the submit. How can I achieve this? Here an example that I want.

```
function Component() {
  const onSubmit = (data) => {
    const ids: string[] = data.ids
    const newData = useQueries({ queries: ids.map(id => ({ queryKey: id, queryFn: () => getData(id) })) })

    // use newData here and then make a mutation
  }
}
```

I know this portion of code is wrong. How is the best way to do that?

**Comment by TkDodo** [comment, marked as answer]
use `queryClient.fetchQuery`

**Reply by carlosriveroib** [reply]
can I create an array of `queryClient.fetchQuery` and a Promise.all it?

**Reply by TkDodo** [reply]
sure
